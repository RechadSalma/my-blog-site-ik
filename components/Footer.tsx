import React from "react";

type Props = {};

export default function Footer({}: Props) {
  return <div className="bg-primary h-100">Footer</div>;
}
