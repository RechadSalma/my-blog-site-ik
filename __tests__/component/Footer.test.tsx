import { render, screen } from "@testing-library/react";
import Footer from "../../components/Footer";

describe("<Footer/>", () => {
  it("Footer component is showing", () => {
    render(<Footer />);

    const { getByText } = screen;

    const divFooter = getByText(/ilshaad kheerdali/i);

    expect(divFooter).toBeInTheDocument();
  });

  it("display the correct current year (new Date()) within the footer", () => {
    render(<Footer />);

    const { getByText } = screen;

    const currentYear = new Date().getFullYear();

    const timeTag = getByText(currentYear);

    expect(timeTag).toBeInTheDocument();
  });

  it("Ensure all 4 links are showing (github/linkedin/portfolio/email me)", () => {
    const { getByTestId } = render(<Footer />);

    const linksContainer = getByTestId("containerForPersonalLinks");

    expect(linksContainer.children.length).toEqual(4);
  });
});
