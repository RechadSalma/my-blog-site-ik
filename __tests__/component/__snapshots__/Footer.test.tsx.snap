// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> snapshot footer component 1`] = `
<div>
  <small
    class="d-flex flex-column justify-content-center align-items-center h-100 text-light"
    style="font-size: 0.8rem; font-family: 'Montserrat', monospace, 'Roboto', Arial, Helvetica, sans-serif !important;"
  >
    <div>
      Â©
      <time
        datetime="2022"
      >
        2022
      </time>
       Ilshaad Kheerdali
    </div>
  </small>
</div>
`;
